src/Controller/Component/RequestHandlerComponent.php

```diff
@@ -238,7 +239,7 @@ public function convertXml($xml)
      *   set that Content-type in the response header.
      *
      * @param \Cake\Event\EventInterface $event The Controller.beforeRender event.
-     * @return bool false if the render process should be aborted
+     * @return bool|void false if the render process should be aborted
```

Comment:
Response|null actually

Reply:
Fairy certain that is a dead relict from 2.x times.
Just void should be fine for these.

Reply:
It's not a relic. If a listener callback returns `false` the event manager stops event propagation.  `RequestHandler::beforeRender()` checks the last modified headers and if needed stops the event propagation to prevent view rendering and just return response with appropriate headers.

The `return false` could however be replaced with `$event->stopPropgation()` to achieve the same result.